[EDU-4709] edge enhanced arch (#1004)
aziontech/docs: src/content/docs/en/pages/architectures/edge-enhanced/edge-enhanced.mdx

---
title: Enhance Applications with Edge Computing

> description: Learn how to enhance your applications and APIs using Edge Computing.

Reviewer: ```suggestion
description: Learn how to enhance your applications and APIs using edge computing.
```

meta_tags: edge, applications, web applications, api, enhancement
namespace: docs_arch_edge_enhancement
permalink: /documentation/architectures/edge-enhanced/
---

> Edge computing is a technology that brings data storage and processing closer to the source of data - the edge of your network. The use of edge computing introduces several advantages, mainly reduced latency, increased security, and improved scalability. 

Reviewer: ```suggestion
Edge computing is a technology that brings data storage and processing closer to the source of data —the edge of your network. The use of edge computing introduces several advantages, mainly reduced latency, increased security, and improved scalability. 
```

> Using Azion services and products, we demonstrate how an ecommerce application's shopping cart feature can be moved to an edge computing environment. The proposed serverless approach integrated with other edge capabilities offers high performance, scalability, and reliable data transaction.

Reviewer: ```suggestion
Using Azion services and products, it's possible to demonstrate how an e-commerce application's shopping cart feature can be moved to an edge computing environment. The proposed serverless approach integrated with other edge capabilities offers high performance, scalability, and reliable data transactions.
```

## Cart Management at the Edge

> By running the shopping cart at the edge using edge functions, companies can reap several benefits:

Reviewer: ```suggestion
By running the shopping cart at the edge using edge functions, companies can obtain several benefits:
```

> - **Reduced Latency**: since the shopping cart operations are performed closer to the user, the response times are significantly reduced, leading to a smoother and faster user experience.

Reviewer: ```suggestion
- **Reduced latency**: since the shopping cart operations are performed closer to the user, the response times are significantly reduced, leading to a smoother and faster user experience.
```

- **Scalability**: serverless functions scale automatically based on the demand. This means that the shopping cart can handle any number of users without the need for manual intervention or additional resources.

> - **Cost Efficiency**: with serverless, you only pay for the compute time you consume. There is no charge when your code is not running, making it a cost-effective solution for managing shopping carts.

Reviewer: ```suggestion
- **Cost efficiency**: with a serverless approach, you only pay for the computing time you consume. There is no charge when your code isn't running, making it a cost-effective solution for managing shopping carts.
```

> - **Improved Security**: edge computing can provide enhanced security as data is processed closer to its source, reducing the risk of data breaches during transmission.

Reviewer: ```suggestion
- **Improved security**: edge computing can provide enhanced security as data is processed closer to its source, reducing the risk of data breaches during transmission.
```

> - **Cache Utilization**: caching shopping cart data at the edge can further improve performance and reduce costs. It allows for faster retrieval of frequently accessed data, reducing the need for expensive database calls.

Reviewer: ```suggestion
- **Cache utilization**: caching shopping cart data at the edge can improve performance and reduce costs. It allows for faster retrieval of frequently accessed data, reducing the need for expensive database calls.
```

These benefits combined can lead to an overall improved shopping experience for the customers, and more efficient and cost-effective operations for the ecommerce platform.

---

> ## Architecture

Reviewer: ```suggestion
## Enhanced Applications Architecture
```

![Overview of a request/response of shopping cart through Edge Application](/assets/docs/images/uploads/edge-enhanced.png)

### Dataflow

1. The user's cloud-hosted application makes a low-latency request to access a feature running on Azion. This request goes to Azion Traffic Router, which smartly chooses the optimal Edge Node to receive the request and directs the request to this Edge Node.

> 2. The request goes through Edge Firewall and is analyzed for security. If it's not a malicious request, it reaches the Edge Application.

Reviewer: ```suggestion
2. The request goes through Edge Firewall and is analyzed for security. If it isn't a malicious request, it reaches the edge application.
```

3. The request reaches the edge application and based on its efficient Rules Engine configuration, accesses the requested content. If configured, Image Processor, Edge Cache, Application Accelerator, and Edge Functions will be executed. Edge Functions, written in Javascript, run on Azion Edge Runtime, improving the usability and maintainability of decentralized applications.

4. The rule configured to run an edge function is triggered, then the edge function operates the shopping cart logic and can communicate with Edge Storage and Edge SQL, ensuring smooth operations.

5. Operations requiring database transactions are accomplished with Edge SQL. The static content, if any, will be stored in Edge Storage, reducing latency and costs.

After this process, the application returns a response to the request.

---

> ## Components

Reviewer: ```suggestion
## Components

```

    - [Edge Cache](/en/documentation/products/build/edge-application/edge-cache/): global module to cache content at the edge.
    - [Edge Functions](/en/documentation/products/build/edge-application/edge-functions/): allows you to create event-driven, serverless applications, at the edge of the network, closer to users.
    - [Rules Engine](/en/documentation/products/build/edge-application/rules-engine/): a tool to configure the scenarios in which a specific cache policy is enforced.
    - [Tiered Cache](/en/documentation/products/build/edge-application/tiered-cache/): module that creates an additional cache layer between the edge and your origin servers, reducing latency and infrastructure costs. 
    - [Load Balancer](/en/documentation/products/build/edge-application/load-balancer/): module designed to balance traffic across your origins, data centers, or cloud providers, mitigating network congestion and server overload.
    - [Edge Storage](/en/documentation/products/store/edge-storage/): a scalable and secure storage service designed to integrate object storage with the Azion Edge Platform using the S3 standard for object operations.

>     - [Origins](/en/documentation/products/build/edge-application/origins/):  allows you to customize the source address of your Edge Application content, the Host header of your application, the path, and the HMAC authentication credentials for protected services.

Reviewer: ```suggestion
    - [Origins](/en/documentation/products/build/edge-application/origins/): allows you to customize the source address of your Edge Application content, the Host header of your application, the path, and the HMAC authentication credentials for protected services.
```

    - [Domains](/en/documentation/products/build/edge-application/domains/): register a custom domain with Azion to deliver the edge application.
- [Azion Marketplace](/en/documentation/products/marketplace/) is a digital catalog that makes it easy to find, test, and deploy edge-running software.
- [Azion CLI](/en/documentation/products/azion-cli/overview/):  an open-source command-line interface (CLI) that lets you interact with Azion Edge Platform using a terminal. 

> - [Azion Console](/en/documentation/products/guides/getting-to-know-azion-console/): Azion’s configuration interface where you can access all the available products and their settings. It enables you to complete all tasks in Azion’s platform

Reviewer: ```suggestion
- [Azion Console](/en/documentation/products/guides/getting-to-know-azion-console/): Azion’s configuration interface where you can access all the available products and their settings. It enables you to complete all tasks in Azion’s platform.
```

---

## Implementation

1. Create an edge application:
    - On [Console](https://console.azion.com), create an application using the **+ Create** button on the homepage.
    - Using the [Azion API](https://api.azion.com/#d43c5509-fb99-4b20-a7bd-64f0924af352) to create an edge application.
    - Using the [Azion CLI](/en/documentation/devtools/cli/create/#edge-applications) to create and deploy an edge application.
2. Create a domain and associate it with the edge application: 
    - On Console, [create a domain](/en/documentation/products/guides/configure-a-domain/) using the + Create button on the homepage.
    - Using the [Azion API](https://api.azion.com/#d43c5509-fb99-4b20-a7bd-64f0924af352) to create a domain.

>     - If you created an application through a template, the domain is created for you.

Reviewer: na menciona o template como opção antes
